src/swift/Sources/Models/TypeModel.swift

```diff
@@ -31,16 +31,11 @@ struct TypeModel {
     var name: String
     var isOptional = false
     var isArray = false
-
-    init(from source: IdentifierPattern) {
-        guard let typeAnnotation = source.typeAnnotation?.type else {
-            SharedLogger.fail("Missing type annotation.")
-        }
-        self.init(from: typeAnnotation)
-    }
```

Comment:
Can you explain what the TypeModel is doing and why you removed and added different initializers?

Reply:
TypeModel is just of way of controlling the rendering of types in a consistent way. swift-ast has a bunch on incompatible ways of representing types which is frustrating for trying to render, so this allows us to work just with one known type and handle the mapping in one place (via initializers). It's very similar in concept to "view models" using in UI-based applications (we use it extensively in autorest.swift).